testing/test_BumpHunter.py

```diff
@@ -2,79 +2,107 @@
 # Here, all the values required to pass te test has been validated using the
 # orriginal c++ version of BumpHunter used by ATLAS collaboration.
 
-import pyBumpHunter as BH
-import pytest
-import uproot as upr
+import os
+
 import numpy as np
+import pytest
+
+import pyBumpHunter as BH
 
 # Import the data
-File = upr.open('data/data.root')
-bkg = File['bkg'].arrays(library='np')['bkg']
-data = File['data'].arrays(library='np')['data']
-sig = File['sig'].arrays(library='np')['sig']
-
-# Create the BumpHunter instance
-BHtest = BH.BumpHunter(rang=[0,20],
-                       width_min=2,
-                       width_max=6,
-                       width_step=1,
-                       scan_step=1,
-                       Npe=10000,
-                       Nworker=1,
-                       seed=666)
+
+here = os.path.dirname(os.path.realpath(__file__))
+
+
+def make_datasets(seed):
+    np.random.seed(seed)
+    bkg = np.random.exponential(2, size=100000)
+    sig = np.random.normal(5.5, 0.35, size=5000)
+    data = np.concatenate(
+        [np.random.exponential(2, size=100000), np.random.normal(5.5, 0.35, size=150)]
+    )
+    np.random.shuffle(data)  # to make sure peak is not ordered
+
+    all = [ar[ar < 20] for ar in (data, sig, bkg)]
+    return all
+
+
+@pytest.fixture
+def data_sig_bkg1():
+    return make_datasets(seed=534)
+
+
+@pytest.fixture
+def bhunter_noscan():
+    return BH.BumpHunter1D(
+        rang=[0, 20],
+        width_min=2,
+        width_max=6,
+        width_step=1,
+        scan_step=1,
+        Npe=10000,
+        nworker=1,
+        seed=666,
+    )
+
+
+@pytest.fixture
+def bhunter(bhunter_noscan, data_sig_bkg1):
+    data, _, bkg = data_sig_bkg1
+    bhunter_noscan.BumpScan(data, bkg)
+    return bhunter_noscan
+
 
 # Test if the the BumpScan method runs
-def test_scan_run():
-    BHtest.BumpScan(data,bkg)
+def test_scan_run(data_sig_bkg1, bhunter_noscan):
```

Comment:
Important: do not rely on the execution order of your tests! Assume they have a random order

Reply:
You can use pytest fixtures to force setup/teardown if you need to, or share it between tests. Remember a user can just use `-k` to select a subset of tests!